leapdao/leap-node
# fix: epoch length tx replays

Proposed solution:
Don't use `epochLengthIndex` for EpochLengthV2 txs. Instead, use blockHeight from tx to find
corresponding event in the array. For legacy EpochLengthV1 txs use the old logic.

Fixes #405 

How to reproduce:

1. Start a local ganache with contracts (sandbox: `yarn start --onlyRoot`).
2. Start a node connecting to that root chain (you need to copy sandbox's network config as usual)
3. Submit a EpochLength tx to the root chain. I use leap-guardian for this: `SLOT=0 EPOCH_LENGTH=1 NODE_URL=http://localhost:8645 PRIV_KEY=<key from ganache> node scripts/setValidator.js`
4. Wait for event to be relayed (check node logs for EpochLength bridge event)
5. Stop the node and clean the state (run with `--fresh` arg)
6. start again. Expected epoch length is 1 (as set in step 3)
